src/System.ComponentModel.EventBasedAsync/tests/BackgroundWorkerTests.cs

```diff
@@ -88,10 +88,12 @@ public void RunWorkerAsync_NoOnWorkHandler_SetsResultToNull()
             stopwatch.Start();
             while (!isCompleted)
             {
-                if (stopwatch.Elapsed > TimeSpan.FromSeconds(10))
+                if (stopwatch.Elapsed > TimeSpan.FromSeconds(20))
                 {
-                    throw new Exception("The background worker never completed.");
+                    throw new Exception("The background worker did not complete in 20 sec.");
                 }
+
+                Thread.Sleep(50);
```

Comment:
We shouldn't need to spin/block at all (I'm actually surprised this test isn't hanging on a single core machine given xunit's synchronization context and BackgroundWorker's posting back to the captured synchronization context).  It'd be better to have something like:
```C#
public async Task RunWorkerAsync_NoOnWorkHandler_SetsResultToNull()
{
    var tcs = new TaskCompletionSource<bool>();
    var backgroundWorker = new BackgroundWorker { WorkerReportsProgress = true };
    backgroundWorker.RunWorkerCompleted += (sender, e) =>
    {
        Assert.Null(e.Result);
        Assert.False(backgroundWorker.IsBusy);
        tcs.SetResult(true);
    };

    var stopwatch = Stopwatch.StartNew();
    backgroundWorker.RunWorkerAsync();
    await Task.WhenAny(tcs.Task, Task.Delay(10));
    Assert.True(tcs.IsCompleted);
}
```

Reply:
You're right, I was being a bit too minimal. I"ll do that.